src/vizdoom/src/viz_screen.cpp

```diff
@@ -178,7 +178,7 @@ void VIZ_ScreenUpdateSM(){
         SMBufferSize[3] = vizScreenSize;
     }
 
-    VIZ_SMUpdate(SMBuffersSize);
+    VIZ_SMUpdate(SMBuffersSize + buffer_len * sizeof(short) * 2);
```

Comment:
Cmon :)

This should be an `if` increasing the size of SMBuffersSize, like the other cases.
Also `buffer_len` is not a good variable name. Maybe `audioBufferLen`?

Reply:
Yup, fixed